### PR TITLE
Add ch.tlaun.TL

### PR DESCRIPTION
**Pull request by Prototik**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: ch.tlaun.TL.yaml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    build-commands:
+      - install -Dt /app/share/applications ch.tlaun.TL.desktop
+      - install -Dt /app/share/metainfo ch.tlaun.TL.appdata.xml
+      - install -D ch.tlaun.TL.48x48.png /app/share/icons/hicolor/48x48/apps/ch.tlaun.TL.png
+      - install -D ch.tlaun.TL.64x64.png /app/share/icons/hicolor/64x64/apps/ch.tlaun.TL.png
+      - install -D ch.tlaun.TL.96x96.png /app/share/icons/hicolor/96x96/apps/ch.tlaun.TL.png
+      - install -D ch.tlaun.TL.128x128.png /app/share/icons/hicolor/128x128/apps/ch.tlaun.TL.png
+      - install -D ch.tlaun.TL.192x192.png /app/share/icons/hicolor/192x192/apps/ch.tlaun.TL.png
+      - install -D ch.tlaun.TL.256x256.png /app/share/icons/hicolor/256x256/apps/ch.tlaun.TL.png
+command: ch.tlaun.TL
```
Comment (hfiguiere) [inline ch.tlaun.TL.yaml:75]: these three sections are usually above module

File: ch.tlaun.TL.yaml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        path: ch.tlaun.TL.192x192.png
+      - type: file
+        path: ch.tlaun.TL.256x256.png
+    build-commands:
+      - install -Dt /app/share/applications ch.tlaun.TL.desktop
+      - install -Dt /app/share/metainfo ch.tlaun.TL.appdata.xml
+      - install -D ch.tlaun.TL.48x48.png /app/share/icons/hicolor/48x48/apps/ch.tlaun.TL.png
+      - install -D ch.tlaun.TL.64x64.png /app/share/icons/hicolor/64x64/apps/ch.tlaun.TL.png
+      - install -D ch.tlaun.TL.96x96.png /app/share/icons/hicolor/96x96/apps/ch.tlaun.TL.png
+      - install -D ch.tlaun.TL.128x128.png /app/share/icons/hicolor/128x128/apps/ch.tlaun.TL.png
```
Comment (hfiguiere) [inline ch.tlaun.TL.yaml:72]: You only need 128x128. 256x256 is ok too. The rest is unnecessary. A SVG is a good idea if there is one.

File: ch.tlaun.TL.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+    <id type="desktop">ch.tlaun.TL.desktop</id>
+    <metadata_license>CC0-1.0</metadata_license>
+    <name>TL</name>
+    <name_variant_suffix>Legacy</name_variant_suffix>
+    <summary>Play Minecraft and create your own world!</summary>
+    <description>
+        <p>TL is Minecraft Launcher.</p>
```
Comment (hfiguiere) [inline ch.tlaun.TL.appdata.xml:9]: this is very vague as a description.

File: ch.tlaun.TL.appdata.xml [hunk 4]
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+    <id type="desktop">ch.tlaun.TL.desktop</id>
```
Comment (hfiguiere) [inline ch.tlaun.TL.appdata.xml:3]:
```suggestion
    <id type="desktop">ch.tlaun.TL</id>
```

the app-id shall not end with .deskop.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: ch.tlaun.TL.appdata.xml (hunk 4 above)
Comment (Prototik) [reply ch.tlaun.TL.appdata.xml:3]: Actually it should, otherwise flatpak-builder will fail with error.

File: ch.tlaun.TL.yaml (hunk 1 above)
Comment (Prototik) [reply ch.tlaun.TL.yaml]: fixed

File: ch.tlaun.TL.yaml (hunk 2 above)
Comment (Prototik) [reply ch.tlaun.TL.yaml:73]: fixed

File: ch.tlaun.TL.appdata.xml (hunk 3 above)
Comment (Prototik) [reply ch.tlaun.TL.appdata.xml]: fixed

File: ch.tlaun.TL.appdata.xml (hunk 4 above)
Comment (hfiguiere) [reply ch.tlaun.TL.appdata.xml:3]: I doubt it fail because of that.

I must insist.

```suggestion
    <id>ch.tlaun.TL</id>
```

Comment (Prototik) [reply ch.tlaun.TL.appdata.xml]: Okay, I wrongly checked `<launchable>` tag, sorry.
Fixed!

**Comment by Prototik** [comment]
bot, build ch.tlaun.TL

**Comment by hfiguiere** [comment]
```
builddir/files/share/appdata/ch.tlaun.TL.appdata.xml: FAILED:
• style-invalid         : Not enough <screenshot> tags, minimum is 1
• style-invalid         : <name> is too short [TL] minimum is 3 chars
Validation of files failed
```

**Comment by Prototik** [comment]
bot, build ch.tlaun.TL

**Comment by Prototik** [comment]
@hfiguiere seems okay for me, could you review it please?

File: ch.tlaun.TL.appdata.xml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        <content_attribute id="language-humor">none</content_attribute>
+        <content_attribute id="language-discrimination">none</content_attribute>
+        <content_attribute id="social-chat">intense</content_attribute>
+        <content_attribute id="social-info">none</content_attribute>
+        <content_attribute id="social-audio">none</content_attribute>
+        <content_attribute id="social-location">none</content_attribute>
+        <content_attribute id="social-contacts">none</content_attribute>
+        <content_attribute id="money-purchasing">none</content_attribute>
+        <content_attribute id="money-gambling">none</content_attribute>
+    </content_rating>
```
Comment (barthalion) [inline ch.tlaun.TL.appdata.xml:56]: Fields set to `none` can be removed.

**Comment by barthalion** [comment]
Hey @pizzalovingnerd, does that launcher do anything that would be considered illegal, like it was with TLauncher?

File: ch.tlaun.TL.appdata.xml (hunk 5 above)
Comment (Prototik) [reply ch.tlaun.TL.appdata.xml]: fixed

**Comment by barthalion** [comment]
My understanding is that TL allows to install and launch Minecraft without verifying any accounts like other launchers on Flathub do. Based on that, it cannot be published to Flathub.

**Comment by Prototik** [comment]
@barthalion we don't force users to use any kind of cracked accounts. We support both legal microsoft accounts and alternative auth systems, required by some third-party servers.

If we make special version of TL, which will require at least one legal microsoft account to use other kinds, will it be suitable for Flathub?

**Comment by barthalion** [comment]
Yes, that would be fine. We already had some drama related to "pirated" launchers, and I'd like to avoid any of it.

**Comment by Prototik** [comment]
bot, build ch.tlaun.TL

**Comment by Prototik** [comment]
@barthalion done, now TL will require Microsoft or Mojang account before any further usage:
image

**Comment by PizzaLovingNerd** [comment]
> Hey @PizzaLovingNerd, does that launcher do anything that would be considered illegal, like it was with TLauncher?

Was pinging me intentional? I'm not affiliated with this project.

**Comment by Prototik** [comment]
@barthalion any progress?

**Comment by barthalion** [comment]
bot, build ch.tlaun.TL

**Pull request closed without merging**